pkg/services/udp_proxy_test.go

```diff
@@ -150,3 +153,111 @@ func TestUDPProxyServiceOutbound(t *testing.T) {
+			},
+		},
+		{
+			name:         "Written bytes mismatch",
+			expectedAddr: expectedAddr,
+			calls: func(mockPacketConner *mock_netceptor.MockPacketConner, mockUDPConnInterface *mock_net_interface.MockUDPConnInterface) {
+				mockPacketConner.EXPECT().ReadFrom(gomock.Any()).Return(7, expectedUDPAddr, nil).Times(1)
+				mockUDPConnInterface.EXPECT().WriteTo(gomock.Any(), gomock.Any()).Return(5, nil).Times(1)
+				mockPacketConner.EXPECT().GetLogger().Return(logger.NewReceptorLogger("")).Times(1)
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+			mockPacketConner := mock_netceptor.NewMockPacketConner(ctrl)
+			mockUDPConnInterface := mock_net_interface.NewMockUDPConnInterface(ctrl)
+			tt.calls(mockPacketConner, mockUDPConnInterface)
+			buf := make([]byte, utils.NormalBufferSize)
+			processInboundPacket(mockPacketConner, mockUDPConnInterface, expectedUDPAddr, tt.expectedAddr, buf)
+		})
+	}
+}
+
+func TestProcessOutboundPacket(t *testing.T) {
+	destinationAddr := &net.UDPAddr{IP: net.ParseIP("192.168.0.1"), Port: 2222}
+	tests := []struct {
+		name           string
+		Addr           netceptor.Addr
+		calls          func(mockPacketConner *mock_netceptor.MockPacketConner, mockUDPConnInterface *mock_net_interface.MockUDPConnInterface)
+		expectContinue bool
+	}{
+		{
+			name: "Read error",
+			calls: func(mockPacketConner *mock_netceptor.MockPacketConner, mockUDPConnInterface *mock_net_interface.MockUDPConnInterface) {
+				mockUDPConnInterface.EXPECT().Read(gomock.Any()).Return(1, fmt.Errorf("Read error")).Times(1)
+				mockPacketConner.EXPECT().GetLogger().Return(logger.NewReceptorLogger("")).Times(1)
+			},
+		},
+		{
+			name: "WriteTo error",
+			calls: func(mockPacketConner *mock_netceptor.MockPacketConner, mockUDPConnInterface *mock_net_interface.MockUDPConnInterface) {
+				mockUDPConnInterface.EXPECT().Read(gomock.Any()).Return(1, nil).Times(1)
+				mockPacketConner.EXPECT().WriteTo(gomock.Any(), gomock.Any()).Return(1, fmt.Errorf("WriteTo error")).Times(1)
+				mockPacketConner.EXPECT().GetLogger().Return(logger.NewReceptorLogger("")).Times(1)
+			},
+		},
+		{
+			name: "Written bytes mismatch",
+			calls: func(mockPacketConner *mock_netceptor.MockPacketConner, mockUDPConnInterface *mock_net_interface.MockUDPConnInterface) {
+				mockUDPConnInterface.EXPECT().Read(gomock.Any()).Return(7, nil).Times(1)
+				mockPacketConner.EXPECT().WriteTo(gomock.Any(), gomock.Any()).Return(5, nil).Times(1)
+				mockPacketConner.EXPECT().GetLogger().Return(logger.NewReceptorLogger("")).Times(1)
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+			mockPacketConner := mock_netceptor.NewMockPacketConner(ctrl)
+			mockUDPConnInterface := mock_net_interface.NewMockUDPConnInterface(ctrl)
+			tt.calls(mockPacketConner, mockUDPConnInterface)
```

Comment:
```suggestion
			if tt.calls != nil {
			     tt.calls()
			}
```
Need to add a check if `tt.calls()` is nil. This is to safeguard the tests against panics when we are adding new test cases into the slice.

Reply:
Done, and updated all tests in this file with that. Thank you!